core/src/main/java/me/aurium/branch/nodes/results/SearchInput.java

```diff
@@ -45,6 +45,13 @@ public SearchInput(Deque<Block> initialPath, Deque<Block> reducablePath) {
         return reducablePath;
     }
 
+    /**
+     * TODO: remove or fix this: i'd preferably like the internal ""initial"" deque to be sealed and immutable
+     *  It would be nice to enforce immutability on reducablePath except the fact is the reducable path exists
+     *  to be immutable and edited, so maybe not? Please advise.
```

Comment:
A Deque is intended to be mutated and manipulated. It doesn't make much sense to have an immutable one. For immutable or read-only access I'd suggest a List.

Reply:
The deques will remain mutable then